mockgcp/mockcompute/nodetemplatesv1.go

```diff
@@ -0,0 +1,119 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockcompute
+
+import (
+	"context"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/cloud/compute/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type NodeTemplatesV1 struct {
+	*MockService
+	pb.UnimplementedNodeTemplatesServer
+}
+
+func (s *NodeTemplatesV1) Get(ctx context.Context, req *pb.GetNodeTemplateRequest) (*pb.NodeTemplate, error) {
+	name, err := s.newNodeTemplateName(req.GetProject(), req.GetRegion(), req.GetNodeTemplate())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeTemplate{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node template %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading node template: %v", err)
+		}
+	}
+
+	return obj, nil
+}
+
+func (s *NodeTemplatesV1) Insert(ctx context.Context, req *pb.InsertNodeTemplateRequest) (*pb.Operation, error) {
+	name, err := s.newNodeTemplateName(req.GetProject(), req.GetRegion(), req.GetNodeTemplateResource().GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	id := s.generateID()
+
+	obj := proto.Clone(req.GetNodeTemplateResource()).(*pb.NodeTemplate)
+	obj.SelfLink = PtrTo("https://compute.googleapis.com/compute/v1/" + name.String())
+	obj.CreationTimestamp = PtrTo(s.nowString())
+	obj.Id = &id
+	obj.Kind = PtrTo("compute#nodetemplate")
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating node template: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeTemplatesV1) Delete(ctx context.Context, req *pb.DeleteNodeTemplateRequest) (*pb.Operation, error) {
+	name, err := s.newNodeTemplateName(req.GetProject(), req.GetRegion(), req.GetNodeTemplate())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeTemplate{}
+	if err := s.storage.Delete(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node template %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting node template: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+type nodeTemplateName struct {
+	Project *projects.ProjectData
+	Region  string
+	Name    string
+}
+
+func (n *nodeTemplateName) String() string {
+	return "projects/" + n.Project.ID + "/zones/" + n.Region + "/nodeTemplates/" + n.Name
```

Comment:
I think the round-trip tests will likely flag this one, but I think s/zones/regions/g

https://cloud.google.com/compute/docs/reference/rest/v1/nodeTemplates/get

`GET https://compute.googleapis.com/compute/v1/projects/{project}/regions/{region}/nodeTemplates/{nodeTemplate}`

Reply:
updated